### PR TITLE
Create Graph for DRYness Percentage 

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Problem

We have yet to create visualizations for all of our COCOMO metrics, specifically the DRYness percentage. 

DRYness is a measure of how much a project is adhering to DRY (Don't Repeat Yourself) principles. 

DRYness = ULOC / SLOC

ULOC = Unique Lines of Code
SLOC = Source Lines of Code

More info about DRYness can be found here:

- https://github.com/boyter/scc?tab=readme-ov-file#unique-lines-of-code-uloc
- https://lobste.rs/s/has9r7/uloc_unique_lines_code

## Solution

Implement a visualization of the DRYness percentage using a half pie chart.

## Result
DRYness_dedupliFHIR_data

Summary:

* Add a method to parse the output of the scc dryness option 
* Add a method to parse the dryness values into a pie chart
* Handle edge case where dryness value is NaN
* Added dryness graph to repository metrics template file.
* Fixed minor typo in metrics template file

## Test Plan

I have tested locally.

### CODE REVIEW

File: templates/repo_report_template.md [hunk 1]
```diff
@@ -111,6 +111,8 @@ date_stampLastWeek: {date_stamp}
       {{% assign optionsArray = '1 Month, 6 Month' | split: ',' %}}
       {{% assign graphsArray = '/{repo_owner}/{repo_name}/new_commit_contributors_by_day_over_last_month_{repo_name}_data.svg, /{repo_owner}/{repo_name}/new_commit_contributors_by_day_over_last_six_months_{repo_name}_data.svg' | split: ',' %}}
       {{% render "graph-toggle", baseurl: site.baseurl, name: "new-contributors" options: optionsArray, graphs: graphsArray, title: "Number of Contributors Joining per Interval" %}}
-    <!-- Predominant Lanugages Graph -->
+    <!-- Predominant Languages Graph -->
```
Comment (sachin-panayil) [inline templates/repo_report_template.md:114]: good catch!

File: scripts/gen_graphs.py [hunk 2]
```diff
@@ -230,8 +244,79 @@ def generate_predominant_languages_graph(oss_entity):
...
+        if 'Unique Lines of Code' in line:
+            #Use regex to remove all non-numerals from the string
+            dryness_metrics['total_uloc'] = re.sub('[^0-9.]','',line)
+        if 'DRYness' in line:
+            #Use regex to remove all non-numerals from the string
+            dryness_metrics['DRYness_percentage'] = re.sub('[^0-9.]','',line)
+
+    return dryness_metrics
+
+
```
Comment (sachin-panayil) [inline scripts/gen_graphs.py:281]: remove extra lines here

**Review by sachin-panayil — APPROVED** [review]
great work!
